src/views/EventsView/index.tsx

```diff
@@ -1,26 +1,60 @@
 'use client';
-import EventList from '@/components/EventList';
+import React from 'react';
+import { Box, Button, Link, Tab, Tabs, Typography } from '@mui/material';
 import { EventResponse } from '@/types/dataModel/event';
-import { Box, Button, Typography } from '@mui/material';
-import Link from 'next/link';
+import dayjs from 'dayjs';
+import EventList from '@/components/EventList';
 import SearchField from '@/components/SearchField';
 import useSearch from '@/hooks/useSearch';
 
 interface EventsViewProps {
-  events: EventResponse[];
+  events: EventResponse[]
 }
 
-export default function EventsView({ events }: EventsViewProps) {
-  const search = useSearch();
+//Sort Events Ascending Order
+const sortEventsByDateAsc = (events: EventResponse[]) => {
+  return events.sort((a, b) => {
+    return dayjs(a.date).valueOf() - dayjs(b.date).valueOf();
+  });
+};
 
-  if (search.length > 0) {
-    events = events.filter((event) =>
-      `${event.name}`.toLowerCase().includes(search.toLowerCase())
-    );
-  }
+//Sort Events Descending Order
+const sortEventsByDateDesc = (events: EventResponse[]) => {
+  return events.sort((a, b) => {
+    return dayjs(b.date).valueOf() - dayjs(a.date).valueOf();
+  });
+};
+
+export default function EventsPage({events}: EventsViewProps) {
+    const search = useSearch();
+
+    //Search Filter
+    if (search.length > 0) {
+      events = events.filter((event) =>
+        `${event.name}`.toLowerCase().includes(search.toLowerCase())
+      );
+    }
 
-  return (
-    <Box>
+    //Set Showing Events by Tabs
+    let upcomingEvents: EventResponse[];
+    let pastEvents: EventResponse[];
```

Comment:
you can use `const`. instead of declaring the variables up here, just declare them where you give them a value below, ie `const upcomingEvents = sortEventsByDateAsc(eventsTodayAndAfter)` on line 46